core/src/main/java/app/futured/arkitekt/core/BaseViewModel.kt

```diff
@@ -21,14 +22,22 @@ import kotlin.reflect.KClass
  * send from ViewModel to Activity/Fragment.
  */
 abstract class BaseViewModel<VS : ViewState> : ViewModel(), LifecycleObserver {
-
     abstract val viewState: VS
 
     private var onStartCalled = false
     private val liveEventBus = LiveEventBus<VS>()
 
     private val observers = mutableMapOf<Observer<Any>, LiveData<Any>>()
 
+    val savedStateHandle: SavedStateHandle? get() = internalSavedStateHandle
```

Comment:
Ako asi chápem dôvod toho prečo je to nullable. Ale nepáči sa mi na tom že to je definované tým aký factory použijem. Ono totižto `savedStateHandle?.whatever()` mi príde ako veľmi zvláštne využitie, lebo tým vlastne naznačujeme možnosť že tam savedStateHandle nebude, ale ono to tam bude a príde mi to tým pádom zmatočné a mätúce. Napadá ma buď spraviť

- subclassu pre BaseViewModel ktorá to bude mať nonnullable a od tiaľto to zmazať a nenastavovať
- alebo požadovať nejaký interface pre viemodel ktorý bude developer museť pridať ku každému viewmodelu kde to bude chcieť použiť. Tu vidím aj výhodu toho že priamo vo viemodely budem explicitne vidieť kde to môžem použiť a kde nie.
- alebo tam pridať niečo ako requireSavedStateHandle

Reply:
Nullable to je pro to, že savedStateHandle má poměrně nízkou využitelnost, proto není potřeba aby se s vývojáři museli babrat (přidávat něco do modulu). Buď ho nechceš (naprostá většina připadů) a nebo ho chceš tak prostě využiješ jinou factory a máš ho. Nicméně require tam dám, můžu při tom vyhodit vyjímku která tě nasměruje na využití správné factory

Reply:
Súhlas, asi najlepšie riešenie.